test/e2e/common/k8s.py

```diff
@@ -82,27 +83,69 @@ def create_custom_resource(
     return _api.create_namespaced_custom_object(
         reference.group, reference.version, reference.namespace, reference.plural, custom_resource)
 
+def patch_custom_resource(
+    reference: CustomResourceReference, custom_resource: dict):
+    _api_client = _get_k8s_api_client()
+    _api = client.CustomObjectsApi(_api_client)
+
+    if reference.namespace is None:
+        return _api.patch_cluster_custom_object(
+            reference.group, reference.version, reference.plural, reference.name, custom_resource)
+    return _api.patch_namespaced_custom_object(
+        reference.group, reference.version, reference.namespace, reference.plural, reference.name, custom_resource)
+
+def delete_custom_resource(
+    reference: CustomResourceReference, wait_periods: int = 1, period_length: int = 5):
```

Comment:
Perhaps this should be split into a waiting delete and an asynchronous delete? I could see this slowing down the integration tests if everything is waiting